configure.py

```diff
@@ -1910,6 +1915,7 @@ def shared_lib_uses_symlinks():
         'dash_c': cc.compile_flags,
 
         'cc_lang_flags': cc.cc_lang_flags(),
+        'cc_sysroot': cc.add_sysroot_option + options.with_sysroot_dir,
```

Comment:
If `options.with_sysroot_dir` is unset, the option should not be used at all. Also, if the compiler doesn't support the option, and the user set `--with-sysroot-dir`, configuration should error out. Something like (untested code):

```
def sysroot_option():
    if options.with_sysroot_dir == '':
        return ''
    if cc.add_sysroot_option == '':
       logging.error('This compiler doesn't support --sysroot option')
    return cc.add_sysroot_option + options.with_sysroot_dir
...
    'cc_sysroot': sysroot_option()
```